pkgs/development/compilers/halide/default.nix

```diff
@@ -22,37 +23,44 @@ assert blas.implementation == "openblas" && lapack.implementation == "openblas";
 
 stdenv.mkDerivation rec {
   pname = "halide";
-  version = "18.0.0";
+  version = "19.0.0";
 
   src = fetchFromGitHub {
     owner = "halide";
     repo = "Halide";
     rev = "v${version}";
-    hash = "sha256-BPalUh9EgdCqVaWC1HoreyyRcPQc4QMIYnLrRoNDDCI=";
+    hash = "sha256-0SFGX4G6UR8NS4UsdFOb99IBq2/hEkr/Cm2p6zkIh/8=";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/halide/Halide/commit/ac2cd23951aff9ac3b765e51938f1e576f1f0ee9.diff";
+      hash = "sha256-kHUecmmcz/4wLGAyo7NHHF3gMBDxPvhUH8vSC/n+Pd0=";
+    })
+    (fetchpatch {
+      url = "https://github.com/halide/Halide/commit/59f4fff30f4ab628da9aa7e5f77a7f1bb218a779.diff";
+      hash = "sha256-r+InpY1kDWeqWJUqbynFfwZZmnHG+fo3CVdAUlYCvcA=";
+    })
+  ];
+
   postPatch = ''
-    # See https://github.com/halide/Halide/issues/7785
-    substituteInPlace 'src/runtime/HalideRuntime.h' \
-      --replace '#if defined(__x86_64__) || defined(__i386__) || defined(__arm__) || defined(__aarch64__)
-    #define HALIDE_CPP_COMPILER_HAS_FLOAT16' \
-                '#if defined(__x86_64__) || defined(__i386__)
-    #define HALIDE_CPP_COMPILER_HAS_FLOAT16'
+    substituteInPlace src/runtime/CMakeLists.txt --replace-fail \
+        '-isystem "''${VulkanHeaders_INCLUDE_DIR}"' \
+        '-isystem "''${VulkanHeaders_INCLUDE_DIR}"
+         -isystem "${llvmPackages.clang}/resource-root/include"'
   '';
 
   cmakeFlags = [
-    "-DWARNINGS_AS_ERRORS=OFF"
     "-DWITH_PYTHON_BINDINGS=${if pythonSupport then "ON" else "OFF"}"
-    "-DTARGET_WEBASSEMBLY=OFF"
     # Disable performance tests since they may fail on busy machines
     "-DWITH_TEST_PERFORMANCE=OFF"
     # Disable fuzzing tests -- this has become the default upstream after the
     # v16 release (See https://github.com/halide/Halide/commit/09c5d1d19ec8e6280ccbc01a8a12decfb27226ba)
     # These tests also fail to compile on Darwin because of some missing command line options...
     "-DWITH_TEST_FUZZ=OFF"
     # Disable FetchContent for flatbuffers and use the version from nixpkgs instead
-    "-DFLATBUFFERS_USE_FETCHCONTENT=OFF"
-    "-DPYBIND11_USE_FETCHCONTENT=OFF"
+    "-DHalide_USE_FETCHCONTENT=OFF"
+    "-DHalide_WASM_BACKEND=OFF"
```

Comment:
Could also add `wabt` to `buildInputs`, I think

Reply:
And `flatbuffers` needs to be in `buildInputs` for cross builds. At least that is what I would expect, the build is still running.

Reply:
Nevermind the cross build, it is broken due to the build tool `binary2cpp` being compiled for the host.
Please add `meta.broken = !stdenv.buildPlatform.canExecute stdenv.hostPlatform`.